### DISCUSSION TITLE
Publish data to specific subscriber

### DISCUSSION THREAD

**Discussion by advancenXX** [eProsima/Fast-DDS#4756] [Q&A] [answered]
There are 2 subscriber with 1 publisher, and the publisher wants to send data to one specific subscriber.
Is there any way to fit this scene? Maybe content filter?

**Comment by JesusPoderoso** [comment, marked as answer]
Hi @advancenXX, that scenario differs from the DDS standard, which is subscription agnostic.
There are two alternatives to perform something similar of your proposed scenario, by using either content filter topic or partitions.
[Content filtered topic](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/topic/contentFilteredTopic/contentFilteredTopic.html) can be configured as specified [in this section of the Fast DDS documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/topic/contentFilteredTopic/createContentFilteredTopic.html#) in the subscribers side, or even in the [publisher' side](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/topic/contentFilteredTopic/writerFiltering.html#).
Using this mechanism has two cons depending of the configuration:
- If configured on the subscribers' side, all subscribers receive the published sample, but it gets processed only by the ones that filter it.
- If configured on the publishers' side (only under [these conditions](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/topic/contentFilteredTopic/writerFiltering.html#conditions-for-writer-side-filtering)), only the desired subscriber would receive the sample with [these caveats](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/topic/contentFilteredTopic/writerFiltering.html#discovery-race-condition).

On the other hand, [partitions](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/domain/domainParticipant/partition.html) create logical distinctions (as domains do, but inside the topic). A publisher can be modified to publish on a specific topic partition, but that would imply re-launching discovery phases in every partition change and waiting for all endpoints to be discovered to avoid data loss.
For this reason, content filter topic would be the recommended option.
